apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx

```diff
@@ -0,0 +1,228 @@
+                initialValue: value.value as number,
+                onValueChange: ({ value }) => setValue(String(value)),
+              });
+          }}
+          onPointerLeave={() => {
+            if (value.type !== "unit") return;
+            numericScrubRef.current?.disconnectedCallback();
+            numericScrubRef.current = undefined;
+          }}
+          variant="ghost"
+          size="1"
+          css={{
+            ...CSSGridLeft,
+            ...CSSIconButton,
+            ...(isCurrentBreakpoint && {
+              bc: "$colors$blue4",
+              "&:not(:hover)": {
+                color: "$colors$blue11",
+              },
+            }),
+          }}
+        >
+          {Icon && <Icon />}
+        </IconButton>
+      </Tooltip>
+      <TextField
+        css={CSSTextField}
+        spellCheck={false}
+        value={value.value}
+        onChange={(event: ChangeEvent<HTMLInputElement>) =>
+          setValue(event.target.value)
+        }
+        onFocus={(event: FocusEvent<HTMLInputElement>) => event.target.select()}
+        onKeyDown={(event: KeyboardEvent<HTMLInputElement>) => {
+          if (value.type !== "unit") return;
+          if (!["ArrowUp", "ArrowDown"].includes(event.code)) return;
+          event.preventDefault();
+          let currentValue = value.value;
+          let currentDelta = 1;
+          if (event.shiftKey) currentDelta = 10;
+          if (event.altKey) currentDelta = 0.1;
+          if (event.code === "ArrowUp")
+            currentValue = currentValue + currentDelta;
+          if (event.code === "ArrowDown")
+            currentValue = currentValue - currentDelta;
+          const currentValueAsString =
+            currentValue % 1
+              ? currentValue.toPrecision(
+                  Math.abs(currentValue).toString().indexOf(".") + 2
+                )
+              : String(currentValue);
+          setValue(currentValueAsString);
+        }}
+      />
+      {value.type === "unit" ? (
+        <Units
+          value={value.unit}
+          items={units
+            .filter((v) =>
+              [
+                "%",
+                "px",
+                "rem",
+                "em",
+                "ch",
+                "vh",
+                "vw",
+                "hv",
+                "vmin",
+                "vmax",
+              ].includes(v)
+            )
+            .map((unit) => ({ name: unit, label: unit }))}
+          onChange={(unit) => setValue(value.value + unit)}
+        />
+      ) : (
+        <Items
+          value={value.value}
+          items={styleConfig.items}
+          onChange={(item) => setValue(item)}
+        />
+      )}
+    </Grid>
+  );
+};
+
+const Units = ({
+  value,
+  items,
+  onChange,
+}: {
+  value: Unit;
+  items: StyleConfig["items"];
+  onChange?: (value: string) => void;
+}) => {
+  return (
+    <Box
+      css={{
+        ...CSSGridRight,
+        button: { ...CSSIconButton },
```

Comment:
I am confused how this supposed to work, button is not a CSS property ... what do I miss?

Reply:
that's weird because it doesn't use "& button", not sure if stitches made & optional for nesting like this, because button is in place of a property name, how would stitches know you are describing a nested selector